examples/blend.cpp

```diff
@@ -80,12 +81,82 @@ class Blend
 public:
     bool init(int argc, char** argv)
     {
-        if (argc != 2) {
-            printf("usage: blend xxx.264\n");
+        if (argc < 2) {
+            usage();
             return false;
         }
+
+        const char* optShort = "bom";
+        const struct option optLong[] = {
+            { "flip", required_argument, NULL, 'F' },
+            { "rotate", required_argument, NULL, 'R' },
+            { "ot", required_argument, NULL, 'T' },
+            { "ms", required_argument, NULL, 'S' },
+            {0, 0, 0, 0},
+        };
+        int opt;
+        while ((opt = getopt_long(argc, argv, optShort, optLong, NULL)) != -1) {
+            switch (opt) {
+            case 'b':
+                m_bBlend = true;
+                break;
+            case 'o':
+                m_bOsd = true;
+                break;
+            case 'm':
+                m_bMosaic = true;
+                break;
+            case 'F': {
+                int flip = atoi(optarg);
+                if (flip != 0 && flip != 1) {
+                    printf("invalid flip type\n");
+                    usage();
+                    return false;
+                }
+                m_flipRot |= (flip + VPP_TRANSFORM_FLIP_H);
+                break;
+            }
+            case 'R': {
+                int rot = atoi(optarg);
+                if (rot != 90 && rot != 180 && rot != 270) {
+                    printf("invalid rotation degree\n");
+                    usage();
+                    return false;
+                }
+                m_flipRot |= (1 << (rot / 90 + 1));
+                break;
+            }
+            case 'T': {
+                int thr = atoi(optarg);
+                if (thr < 0 || thr > 255) {
+                    printf("invalid osd threshold\n");
+                    usage();
+                    return false;
+                }
+                m_osdThreshold = thr;
+                break;
+            }
+            case 'S': {
+                int size = atoi(optarg);
+                if (size < 1 || size >256) {
+                    printf("invalid mosaice block size\n");
+                    usage();
+                    return false;
+                }
+                m_mosaicSize = size;
+                break;
+            }
+            default:
+                printf("invalid argument");
+                usage();
+                return false;
+            }
+        }
+        printf("Blend: %d, OSD(%d): %d, Mosaic(%d): %d, Transform: %u\n",
+            m_bBlend, m_osdThreshold, m_bOsd, m_mosaicSize, m_bMosaic, m_flipRot);
+
         m_input.reset(new VppInputDecode);
-        if (!m_input->init(argv[1])) {
+        if (!m_input->init(argv[argc - 1])) {
```

Comment:
better not assume last option is file name. you can use optind as https://github.com/xuguangxin/libyami/blob/master/examples/grid.cpp#L707